common/src/main/java/com/wynntils/models/gearinfo/GearInfoRegistry.java

```diff
@@ -0,0 +1,420 @@
+            if (questName == null) return Optional.empty();
+
+            return Optional.of(WynnUtils.normalizeBadString(questName));
+        }
+
+        private FixedStats parseFixedStats(JsonObject json) {
+            JsonElement healthJson = json.get("health");
+            int healthBuff = healthJson == null ? 0 : healthJson.getAsInt();
+            List<Pair<Skill, Integer>> skillBonuses = parseSkillBonuses(json);
+            JsonElement attackSpeedJson = json.get("attackSpeed");
+            Optional<GearAttackSpeed> attackSpeed = (attackSpeedJson == null)
+                    ? Optional.empty()
+                    : Optional.of(GearAttackSpeed.valueOf(attackSpeedJson.getAsString()));
+
+            List<GearMajorId> majorIds = parseMajorIds(json);
+            List<Pair<DamageType, RangedValue>> damages = parseDamages(json);
+            List<Pair<Element, Integer>> defences = parseDefences(json);
+
+            return new FixedStats(healthBuff, skillBonuses, attackSpeed, majorIds, damages, defences);
+        }
+
+        private List<GearMajorId> parseMajorIds(JsonObject json) {
+            JsonElement majorIdsJson = json.get("majorIds");
+            if (majorIdsJson == null || majorIdsJson.isJsonNull()) return List.of();
+
+            return majorIdsJson.getAsJsonArray().asList().stream()
+                    .map(majorIdName -> getMajorIdFromString(majorIdName.getAsString()))
+                    .filter(Objects::nonNull)
+                    .toList();
+        }
+
+        protected GearMajorId getMajorIdFromString(String majorIdString) {
+            return this.allMajorIds.stream()
+                    .filter(mId -> mId.id().equals(majorIdString))
+                    .findFirst()
+                    .orElse(null);
+        }
+
+        private List<Pair<Skill, Integer>> parseSkillBonuses(JsonObject json) {
+            List<Pair<Skill, Integer>> list = new ArrayList<>();
+            for (Skill skill : Skill.values()) {
+                String skillJsonName = skill.getApiName() + "Points";
+                JsonElement skillJson = json.get(skillJsonName);
+                if (skillJson == null) continue;
+
+                int minPoints = skillJson.getAsInt();
+                if (minPoints == 0) continue;
+
+                list.add(Pair.of(skill, minPoints));
+            }
+
+            // Return an immutable list
+            return List.copyOf(list);
+        }
+
+        private List<Pair<DamageType, RangedValue>> parseDamages(JsonObject json) {
+            List<Pair<DamageType, RangedValue>> list = new ArrayList<>();
+            // First look for neutral damage, which has a non-standard name
+            JsonElement neutralDamageJson = json.get("damage");
+            if (neutralDamageJson != null) {
+                String rangeString = neutralDamageJson.getAsString();
+                RangedValue range = RangedValue.fromString(rangeString);
+                if (!range.equals(RangedValue.NONE)) {
```

Comment:
Can you unify this so you continue "early" like the code below?

Reply:
Like in extracting common functionality between the neutral and elemental damages? I can try and do that.

The neutral damage check is not a loop so there is no corresponding keyword to `continue` to break out of the if statement, if that was what you meant.

Reply:
Oh, right. You can see I was getting tired 😅. Code is fine, no need to extract stuff to save 2 lines or so.